app/controllers/triannon/application_controller.rb

```diff
@@ -1,4 +1,55 @@
 module Triannon
   class ApplicationController < ActionController::Base
+
+
+    #--- Authentication methods
+    #
+    # The #access_token_data method is generally available to the
+    # application.  The #access_token_generate and #access_token_validate?
+    # methods are consolidated here to provide a unified view of these methods,
+    # although the application generally may not need to call them.  They are
+    # used specifically in the auth_controller and these methods are tested
+    # in the auth_controller_spec.
+
+    # construct and encrypt an access token, using login data
+    # save the token into session[:access_token]
+    def access_token_generate(data)
+      timestamp = Time.now.to_i.to_s # seconds since epoch
+      token = "#{data.to_json};;;#{timestamp}"
+      salt  = SecureRandom.random_bytes(64)
+      key   = ActiveSupport::KeyGenerator.new(timestamp).generate_key(salt)
+      crypt = ActiveSupport::MessageEncryptor.new(key)
+      session[:access_key] = key
+      session[:access_token] = crypt.encrypt_and_sign(token)
```

Comment:
`AS::MessageEncryptor` uses an internal serializer, so you could encrypt and sign a data structure instead of a string, e.g. an Array or Hash:

```
code = crypt.encrypt_and_sign([data, timestamp])
data, timestamp = crypt.decrypt_and_verify(code)
```